### ISSUE TITLE
[filelog receiver] "force_flush_period" not working as expected

### ISSUE THREAD

**Issue by SamerJ** [open-telemetry/opentelemetry-collector-contrib#20867]
### Component(s)

filelog receiver

## Description
I have a filelog receiver set up to read files under a certain directory. The filelog receiver is configured to start reading from the start and it does so. The log lines are successfully read and then exported to a file. The problem lies when I add a log line or two.
I am adding a log line to one file and waiting for the data to be exported, but it never does. When I try to gracefully kill the OTEL collector it exports the added log line to the file.
When adding a lot of log lines (say 100), I can see that the data is exported. The problem is when one log line is added and then nothing happens for a while (or forever).

## Steps to Reproduce
I created a K8s Pod that has 2 containers. 1 container running Linux (Red Hat) that writes log files. Another container with OTEL collector that reads the log files and writes the result into a file "outputOfReadLogsOverHERE.core".

See the yml file attached below. Running it is enough to reproduce the issue. P.S You need to wait at least a minute for all the log lines to be written.
[k8s-issue-reproduced.zip](https://github.com/open-telemetry/opentelemetry-collector-contrib/files/11207720/k8s-issue-reproduced.zip)

## Expected Result
I am expecting that after the "force_flush_period", the read log line will be exported.

Two added log lines "added message" are added to a log file. I am expecting to see the log lines within the file holding the exported data. The file that holds the exported data is named "outputOfReadLogsOverHERE.core".

## Actual Result
If one or two log line are added, the data is never getting exported.

Thanks in advance and please let me know if more information is needed from me.

### Collector version

v0.75.0

## Environment
Using the OTEL Collector Contribution container image v0.75.0 (latest)
ghcr.io/open-telemetry/opentelemetry-collector-releases/opentelemetry-collector-contrib

### OpenTelemetry Collector configuration

```yaml
receivers:
      filelog:
        include: [ /logs/**/*.* ]
        start_at: 'beginning'
        include_file_name: false
        include_file_path_resolved: true
        force_flush_period: 300ms # Explicitly set (same issue when not set)
```

### Additional context

Whether I explicitly set the "force_flush_period" or not, I get the same issue. I have a simple file exporter that is dumping everything into a file.

**Comment by djaglowski** [comment]
@SamerJ, it's possible that the behavior you are seeing is due to the file exporter buffering the data indefinitely. This behavior has been noted in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18251, and hopefully will be resolved by #20675. Do you mind trying the workaround noted [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/18251#issuecomment-1414497442)?

**Comment by SamerJ** [comment]
Hello @djaglowski,
I tested out the suggested fix and indeed the buffering was on the exporter side. I also tried using v0.69.0 which didn't have a buffer for the file exporter and it worked.
I was wrong on diagnosing the issue will just close this issue.
Thanks for the clarification.
